### ISSUE TITLE
Calls not connecting: Element does not handle events of type 'm.call.select_answer'

### ISSUE THREAD

**Issue by Aq32** [vector-im/element-android#2719]
**Describe the bug**
Calls do not work when receiving or sending. 

**To Reproduce**
1. Receive call request
2. Accept call request
3. Stuck at: Call connecting...
4. Manually give up on call and hang up
5. Read chat window

**Expected behavior**
Calls connect sending/receiving.

**Smartphone (please complete the following information):**
 - OS: Android 9

**Additional context**
 - 1.0.13 - Google Play Store
 - Homeserver: matrix.org

In the chat window this is shown when receiving a call:
```
Contact placed a voice call.
Contact sent data to setup the call.
Contact sent data to setup the call.
You answered the call.
You sent data to setup the call.
Element does not handle events of type 'm.call.select_answer'
You ended the call.
```

In the chat window this is shown when sending a call:
```
You placed a voice call
You sent data to setup the call.
Contact ended the call.
Contact ended the call.
```

I'm unclear on whether this error occurred because a 3rd party app is now responsible for Android phone calls or if this is relevant but could be worth mentioning. It's unclear if the problem is only on my end if sending calls doesn't trigger the `Element does not handle events of type 'm.call.select_answer'` when sending a call request rather than receiving.

**Comment by SimonBrandner** [comment]
Duplicate of [#2704](https://github.com/vector-im/element-android/issues/2704)

**Comment by Aq32** [comment]
I wasn't sure if it was the same but was expecting to have this issue and that one was linked if it's actually the same.

**Comment by biastogit** [comment]
same issue here. Android fails to establish connection. The same server with turn is used without any issue between desktop-desktop and desktop-webRTC clients

**Comment by faaaaabi** [comment]
I can also confirm that Desktop<->Desktop and Android<->Android calls work without any problems, but Desktop<->Android calls are not established

**Comment by faaaaabi** [comment]
The Versions are 1.7.17 for Desktop and 1.0.14 for Android

**Comment by jryans** [comment]
As far as I can see, this is because of a Element Web bug (https://github.com/vector-im/element-web/issues/16239) which is now fixed on develop.element.io and will be released momentarily.

**Comment by luntik2012** [comment]
still doens't work

1.3.3 [40103030] (F-651e3478)

**Comment by TNTBOMBOM** [comment]
I have this issue on debian and android, element-desktop (tried nightly as well same thing) version 1.11.0

**Comment by F-Node-Karlsruhe** [comment]
In my case it's only mobile -> mobile that does not work. Mobile -> desktop and the other way round works perfectly fine.

**Comment by sigma2017** [comment]
Just installed the server on ubuntu 20.04 with no gui, chat between 2 android phones works just fine but no call either audio or video. Disappointed :|

**Comment by deepakvamsi** [comment]
@sigma2017  It works in same network because no firewalls or port blocks  are present in between them  blocked and it will work seamlessly .
`try this for setting voip server for matrix
https://matrix-org.github.io/synapse/v1.37/turn-howto.html`

**Comment by EhsanSadeghiMohaseli** [comment]
> voip

i have same problem, calls are work in same network but they can't work in 2 Different network

**Comment by hast0011** [comment]
having problems with calls as well. Sometimes a call works. Sometimes strange things happen. i.e. signaling never stops on a second device when the call is not accepted. Following calls can not be accepted then and they are not signaled properly.

**Comment by TNTBOMBOM** [comment]
@SimonBrandner any idea? ^

**Comment by sigma2017** [comment]
I think you have something wrong configured on the signaling part. I use a second virtual machine with coturn which accepts connections only from my matrix server. Also I have configured it with a Zero SSL certificate, not LetsEncrypt as suggested into the Matrix official documentation.
On the other hand, in the meantime I have switched to SchildiChat which is an element android fork and is working just fine (in may humble oppinion is better than element on android).
The only issue with the calls is only on element android, on iphone it works as expected.
